test/tests/p5.PolySynth.js

```diff
@@ -18,15 +18,21 @@ describe('p5.PolySynth', function () {
 
     expect(polySynth._voicesInUse.getValueAtTime(noteTriggerTime)).to.equal(0);
 
-    polySynth.play('A2', 0, 0, noteDuration);
+    await polySynth.play('A2', 0, 0, noteDuration);
     expect(polySynth._voicesInUse.getValueAtTime(noteActiveTime)).to.equal(1);
-    polySynth.play('A3', 0, 0, noteDuration);
-    expect(polySynth._voicesInUse.getValueAtTime(noteActiveTime)).to.equal(2);
-    polySynth.play('A4', 0, 0, noteDuration);
-    expect(polySynth._voicesInUse.getValueAtTime(noteActiveTime)).to.equal(3);
-
-    expect(polySynth._voicesInUse.getValueAtTime(noteDoneTime)).to.equal(0);
-
-    polySynth.dispose();
+    await polySynth.play('A3', 0, 0, noteDuration);
+    expect(
+      await polySynth._voicesInUse.getValueAtTime(noteActiveTime)
+    ).to.equal(2);
+    await polySynth.play('A4', 0, 0, noteDuration);
+    expect(
+      await polySynth._voicesInUse.getValueAtTime(noteActiveTime)
+    ).to.equal(3);
+
+    expect(await polySynth._voicesInUse.getValueAtTime(noteDoneTime)).to.equal(
+      0
+    );
+
+    await polySynth.dispose();
```

Comment:
I have the same question about all of these `await` operators - I don't think these methods return a Promise and I haven't seen `await` used on methods that don't return a Promise before so I'm not sure the await operator is helping us here. Let me know if I'm missing something though!

Reply:
Hey @therewasaguy, I agree that using await here has no effect. But the problem is that sometimes(like 1 in 4/5) these tests fail as the code is not synchronous. For example,
<img width="489" alt="Screenshot 2021-06-16 at 5 21 35 PM" src="https://user-images.githubusercontent.com/40578313/122214083-4f595380-cec7-11eb-98f4-434253e2d1e2.png">
A quick fix that I can think of is that we layer every statement here with a setTimeout, which would work but will not look good and is inefficient. Can you suggest any other approach or a workaround?

Reply:
this is a tough one, because [the Web Audio clock is separate (and more precise) than the JS clock](https://www.html5rocks.com/en/tutorials/audio/scheduling/).

I think using `getValueAtTime` (which is not built into the Web Audio API, but comes from using a [Tone.JS](https://tonejs.github.io/docs/14.7.77/Signal#getValueAtTime) Signal and is based on the Web Audio clock) is a good approach. But it only works for parameters that utilize the Tone.JS Signal.

I wonder if the reason it fails sometimes is that we don't pass a raw AudioContext time to PolySynth.play (or to any methods). Instead, we pass a "time from now" of zero. "now" is calculated within the play method here
https://github.com/processing/p5.js-sound/blob/a74dd7eadba27a1e27c46cc7b523faf1b2867dc5/src/polysynth.js#L254 and I wonder how different it is from the "now" that we use in the test?

Reply:
for parameters that don't have `getValueAtTime` I think `setTimeout` is reasonable

Reply:
Yeah, 
while exploring that "now" can be different in different cases, I found out that
the "now" in polysyth.js(noteAttack) and the "now" in the test cases is actually coming out to be same as the function is being called while tests are being executed.
But, the time is being different in between the running of tests sometimes, hence sometimes failing the test.
I am adding a commit to this PR which fixes this bug :)

Reply:
yes @satyasaibhushan , interesting solution there in part-4! :shipit: